erpnext/controllers/selling_controller.py

```diff
@@ -548,3 +548,48 @@ def update_project_billing_and_sales(self):
 			doc.set_gross_margin(update=True)
 			doc.set_status(update=True)
 			doc.notify_update()
+
+	@frappe.whitelist()
+	def set_rate_as_cost(self):
```

Comment:
Throw if "Restrict Amounts in Stock Reports to Role" is not permitted for the session user